OWML.ModHelper.Menus/LayoutManager.cs

```diff
@@ -0,0 +1,119 @@
+            LayoutGroup.transform.localPosition = Vector3.zero;
+            ((RectTransform)LayoutGroup.transform).pivot = NormalPivot;
+            UpdateState();
+        }
+
+        public void UpdateState()
+        {
+            var currentTexts = LayoutGroup.gameObject.GetComponentsInChildren<Text>();
+            _styleApplier.SetTexts(currentTexts);
+            Graphic[] currentGraphics = currentTexts;
+            Array.Resize<Graphic>(ref currentGraphics, currentTexts.Length + _constantGraphics.Count);
+            int i = currentTexts.Length;
+            foreach (var graphic in _constantGraphics)
+            {
+                currentGraphics[i] = graphic;
+                i++;
+            }
+            _styleApplier.SetForeground(currentGraphics);
+        }
+
+        public void Clear()
+        {
+            var childCount = LayoutGroup.transform.childCount;
+            for (var i = childCount - 1; i >= 0; i--)
+            {
+                if (!(_constantGraphics.Contains(LayoutGroup.transform.GetChild(i).gameObject.GetComponent<Graphic>())
+                    || _backingGraphics.Contains(LayoutGroup.transform.GetChild(i).gameObject.GetComponent<Graphic>())))
+                {
+                    LayoutGroup.transform.GetChild(i).gameObject.SetActive(false);
+                    GameObject.Destroy(LayoutGroup.transform.GetChild(i).gameObject);
+                }
+            }
+        }
+
+        public void AddText(string text)
+        {
+            AddTextAt(text, LayoutGroup.transform.childCount);
+        }
+
+        public void AddTextAt(string text, int index)
+        {
+            var textObject = new GameObject("Text", new Type[] { typeof(RectTransform) });
+            var textComponent = textObject.AddComponent<Text>();
+            textComponent.text = text;
+            textComponent.fontSize = FontSize;
+            textComponent.font = _styleManager.GetMenuFont();
+            textComponent.color = _styleManager.GetButtonForegroundMenuColor(UIElementState.NORMAL);
+            textComponent.alignment = TextAnchor.MiddleCenter;
+            textObject.AddComponent<LayoutElement>();
+            textObject.transform.SetParent(LayoutGroup.transform);
+            textObject.transform.localScale = _scale;
+            ((RectTransform)textObject.transform).sizeDelta = new Vector2(textComponent.preferredWidth, ((RectTransform)textObject.transform).sizeDelta.y);
+            ((RectTransform)textObject.transform).pivot = NormalPivot;
+            textObject.transform.SetSiblingIndex(index);
+        }
+
+        public void AddPicture(Texture2D texture, float scale = 1.0f)
+        {
+            AddPictureAt(texture, LayoutGroup.transform.childCount, scale);
+        }
+
+        public void AddPictureAt(Texture2D texture, int index, float scale = 1.0f)
+        {
+            var keySprite = Sprite.Create(texture, new Rect(0f, 0f, (float)texture.width, (float)texture.height), NormalPivot);
```

Comment:
If I remember correctly, texture.width and .height are already floats

Reply:
Wrong, but VS indeed tells that explicit cast is redundant